### PR TITLE
Generic performance update

### PR DESCRIPTION
**Pull request by XmYx**
This PR includes many removals of redundant np array - torch tensor conversions, and generally aims to move more operations to the GPU. Reduces the torch cuda collect calls in the depth models, and ensures whichever depth model works nominally in fp16, is casted to. Refactored Adabins inference with the above in mind. Adds Depth-Anything option as a depth model to use via Transformers - albeit it is slower then the Midas so far, but may be compilable, or offload turned off, etc.
On top of that, add's the webui extensions pre gen color matching correctly, as well, as the webui's own post gen colorCorrectionFactor based color correction.
An other minor change is in the sampler, where we now directly use the VAE, without tiling.
Onediff's VAE boost was making the VAE slower, turned it off.

### CODE REVIEW

File: src/deforum/generators/deforum_flow_generator.py [hunk 1]
```diff
@@ -122,6 +123,57 @@ def filter_flow(flow,
...
+
+    Returns:
+        flow: Computed optical flow (numpy array).
+    """
+    # Convert images to grayscale
+    i1_gray = cv2.cvtColor(i1, cv2.COLOR_BGR2GRAY)
+    i2_gray = cv2.cvtColor(i2, cv2.COLOR_BGR2GRAY)
+
+    # Convert images to PyTorch tensors
+    i1_tensor = torch.from_numpy(i1_gray).half().to('cuda').unsqueeze(0).unsqueeze(0)
```
Comment (rewbs) [inline src/deforum/generators/deforum_flow_generator.py:142]: is ignoring `prev_flow` a deliberate part of the optimisation? 

(Note: I don't actually understand how prev_flow is used in the other algorithms  – I just know it is passed in and appears in the code  :)  )

Comment (XmYx) [reply src/deforum/generators/deforum_flow_generator.py:142]: we have 3 new flow algos, two does consider it, one doesnt, all of them yield, different, sometimes artistic effects, and are very quick. they are [Farneback Torch, DIS Medium Torch, Custom]

File: src/deforum/generators/deforum_noise_generator.py [hunk 2]
```diff
@@ -1,39 +1,129 @@
+# import math
```
Comment (rewbs) [inline src/deforum/generators/deforum_noise_generator.py:1]: Remove commented out code or add a title comment explaining why it's useful to keep.

Comment (XmYx) [reply src/deforum/generators/deforum_noise_generator.py]: removed

**Review by rewbs — APPROVED** [review]

**Comment by deforum** [comment]
Optimizations are not mature enough in my eyes to force - with more testing I'd be happy to turn it on by default. Currently optimizations are not "strictly better" they are still situational.

**Pull request merged**